src/v/cluster/controller_backend.cc

```diff
@@ -496,12 +496,39 @@ controller_backend::calculate_learner_initial_offset(
      * Initial learner start offset only makes sense for partitions with cloud
      * storage data
      */
+    if (auto topic_cfg = p->get_topic_config();
+        topic_cfg.has_value() && topic_cfg->get().is_internal()) {
```

Comment:
nit: I was thinking  is_user_topic(topic_namespace_view tp_ns) (seems a bit more stricter than is_internal).

Reply:
Isn't the behavior we're trying to capture tied to whether TS is enabled? There are non-user topics (schemas, audit, transforms) that correctly have TS enabled. Are we sure we want to change this behavior for these topics?